src/ocean/core/UnitTestRunner.d

```diff
@@ -58,8 +58,30 @@ import ocean.transition;
 
 import ocean.stdc.string: strdup, strlen, strncmp;
 import core.sys.posix.unistd: unlink;
-import core.sys.posix.libgen: basename;
-import core.sys.posix.sys.time: gettimeofday, timeval, timersub;
+
+import core.sys.posix.sys.time: gettimeofday, timeval;
+
+static if (__VERSION__ >= 2000 && __VERSION__ < 2073)
+{
+    extern(C) char* basename(char*);
+
+    extern (D): pure: @safe: @nogc: nothrow:
```

Comment:
Are you sure this only applies to the *body* of the `static if` and not all the rest of the module? Also, you can put only one `:` at the end, you don't have to put one after each attribute.

Also, `extern (D)` is redundant here, the previous `extern (C)` only applies to the symbol you are defining.